### PR TITLE
fix(emqx_management): delete files after trace log download

### PR DESCRIPTION
**Pull request by olcai**
When trace logs are downloaded, a zip file is created containing all requested trace logs. After the client has downloaded the zip file, the gathered files need to be deleted. This PR implements file deletion and also creates a unique directory so that concurrent downloads doesn't interfere with each other.

Fixes EMQX-8470.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [X] Added tests for the changes
- [X] Changed lines covered in coverage report
- [X] Change log has been added to `changes/<version>/(feat|fix)-<PR-id>.en.md` and `.zh.md` files
- [X] For internal contributor: there is a jira ticket to track this change
- [ ] ~~If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up~~
- [ ] ~~Schema changes are backward compatible~~

### CODE REVIEW

File: changes/v5.0.15/feat-9781.zh.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Trace files were left on a node when creating a zip file for download. They are now removed when the file is sent. Also, concurrent downloads will no longer interfere with each other.
```
Comment (olcai) [inline changes/v5.0.15/feat-9781.zh.md:1]: Translation needed

File: changes/v5.0.15/fix-9781.zh.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Trace files were left on a node when creating a zip file for download. They are now removed when the file is sent. Also, concurrent downloads will no longer interfere with each other.
```
Comment (lafirest) [inline changes/v5.0.15/fix-9781.zh.md:1]: 当下载 `日志追踪` 的日志时，一些中间文件将存留在处理节点上，现在这个问题得到了修复。同时，并发下载日志将不再相互干扰。

Comment (olcai) [reply changes/v5.0.15/fix-9781.zh.md]: Thank you!

**Review by lafirest — APPROVED** [review]

File: apps/emqx_management/src/emqx_mgmt_api_trace.erl [hunk 3]
```diff
@@ -461,16 +462,26 @@ download_trace_log(get, #{bindings := #{name := Name}, query_string := Query}) -
...
+                    SessionId = emqx_misc:gen_id(),
+                    ZipDir = filename:join([emqx_trace:zip_dir(), SessionId]),
+                    ok = file:make_dir(ZipDir),
+                    %% Write files to ZipDir and create an in-memory zip file
                     Zips = group_trace_file(ZipDir, TraceLog, TraceFiles),
-                    FileName = binary_to_list(Name) ++ ".zip",
-                    ZipFileName = filename:join([ZipDir, FileName]),
-                    {ok, ZipFile} = zip:zip(ZipFileName, Zips, [{cwd, ZipDir}]),
+                    ZipName = binary_to_list(Name) ++ ".zip",
+                    {ok, {ZipName, Binary}} = zip:zip(ZipName, Zips, [memory, {cwd, ZipDir}]),
```
Comment (thalesmg) [inline apps/emqx_management/src/emqx_mgmt_api_trace.erl:474]: Nit: just to be sure:

```suggestion
                    try
                        {ok, {ZipName, Binary}} = zip:zip(ZipName, Zips, [memory, {cwd, ZipDir}])
                    after
                        ok = file:del_dir_r(ZipDir)
                    end,
```

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_management/src/emqx_mgmt_api_trace.erl (hunk 3 above)
Comment (olcai) [reply apps/emqx_management/src/emqx_mgmt_api_trace.erl]: Thank you, this has been fixed now.

**Review by thalesmg — APPROVED** [review]

**Pull request merged**